src/es-module-shims.js

```diff
@@ -386,12 +389,31 @@ function processPreload (link) {
   link.ep = true;
   // prepopulate the load record
   const fetchOpts = getFetchOpts(link);
-  // save preloaded fetch options for later load  
+  // save preloaded fetch options for later load
   fetchOptsMap.set(link.href, fetchOpts);
   fetch(link.href, fetchOpts);
 }
 
-function resolve (id, parentUrl) {
+
+async function resolveHook(id, parentUrl) {
```

Comment:
Can we just call this `resolve`. I actually misread initially, I think just the rename clarifies my original confusion.